### PR TITLE
5.1.0 release notes: add the changes to committer-concurrency

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/23776
https://github.com/pingcap/docs-cn/pull/7142

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]

File: releases/release-5.0.1.md [hunk 1]
```diff
@@ -8,6 +8,12 @@ title: TiDB 5.0.1 Release Notes
 
 TiDB 版本：5.0.1
 
+## 兼容性更改
```
Comment (youjiali1995) [inline releases/release-5.0.1.md:11]: 这不算兼容性更改吧

Comment (youjiali1995) [reply releases/release-5.0.1.md:11]: 倾向于放在改进提升里

Comment (qiancai) [reply releases/release-5.0.1.md:11]: 之前看到的配置项 默认值变更也写到了 release notes 的兼容性变更（例如以下）里。关于默认值变化，目前对应的文档里没写，如果 release notes 也不写，可能用户就无法知道这个变更了。

- https://docs.pingcap.com/zh/tidb/stable/release-5.0.0#%E9%85%8D%E7%BD%AE%E6%96%87%E4%BB%B6%E5%8F%82%E6%95%B0
https://docs.pingcap.com/zh/tidb/stable/release-5.1.0

- https://docs.pingcap.com/zh/tidb/stable/release-5.1.0#%E9%85%8D%E7%BD%AE%E6%96%87%E4%BB%B6%E5%8F%82%E6%95%B0

**Review by youjiali1995 — APPROVED** [review]

**Pull request merged**